### PR TITLE
Ergodox Community keyboards “coderkun”

### PR DESCRIPTION
**Pull request by coderkun**
Fix the Ergodox Community keyboard “coderkun_neo2” by including the German keymap header file.

Add the Ergodox Community keyboard “coderkun_neo3” for Bone/Neo3.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

### CODE REVIEW

File: layouts/community/ergodox/coderkun_neo3/keymap.c [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+                register_code(KC_RSFT);
+            }
+            else {
+                unregister_code(KC_RSFT);
+            }
+        break;
+    }
+
+    return MACRO_NONE;
+};
```
Comment (zvecr) [inline layouts/community/ergodox/coderkun_neo3/keymap.c:241]: This method of macros is deprecated, and should be updated to https://docs.qmk.fm/#/feature_macros for it to be accepted.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: layouts/community/ergodox/coderkun_neo3/keymap.c (hunk 1 above)
Comment (coderkun) [reply layouts/community/ergodox/coderkun_neo3/keymap.c:241]: Thanks for the link, I will look into it.

File: layouts/community/ergodox/coderkun_neo3/keymap.c [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+// Enable LED based on "lock" state
+void led_set_user(uint8_t usb_led) {
+    if (usb_led & (1<<USB_LED_SCROLL_LOCK)) {
+        ergodox_right_led_1_on();
+    }
+    else {
+        ergodox_right_led_1_off();
+    }
+};
```
Comment (drashna) [inline layouts/community/ergodox/coderkun_neo3/keymap.c:281]:
```suggestion
// Enable LED based on "lock" state
bool led_update_user(led_t led_state) {
    if (led_state.scroll_lock) {
        ergodox_right_led_1_on();
    } else {
        ergodox_right_led_1_off();
    }
    return true;
};
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: layouts/community/ergodox/coderkun_neo3/keymap.c (hunk 1 above)
Comment (coderkun) [reply layouts/community/ergodox/coderkun_neo3/keymap.c]: I fixed this by removing the macro.

File: layouts/community/ergodox/coderkun_neo2/keymap.c [hunk 3]
```diff
@@ -1,116 +1,115 @@
 #include QMK_KEYBOARD_H
 #include "action_layer.h"
```
[the remark this answers is not in the document]
Comment (coderkun) [reply layouts/community/ergodox/coderkun_neo2/keymap.c]: Fixed.

File: layouts/community/ergodox/coderkun_neo3/keymap.c [hunk 4]
```diff
@@ -0,0 +1,320 @@
+#include QMK_KEYBOARD_H
+#include "action_layer.h"
```
[the remark this answers is not in the document]
Comment (coderkun) [reply layouts/community/ergodox/coderkun_neo3/keymap.c]: Fixed.

File: layouts/community/ergodox/coderkun_neo3/keymap.c [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+    ergodox_board_led_off();
+    ergodox_right_led_1_off();
+    ergodox_right_led_2_off();
+    ergodox_right_led_3_off();
+};
+
+
+// Runs constantly in the background, in a loop.
+void matrix_scan_user(void) {
+    uint8_t layer = biton32(layer_state);
```
[the remark this answers is not in the document]
Comment (coderkun) [reply layouts/community/ergodox/coderkun_neo3/keymap.c]: Fixed.

File: layouts/community/ergodox/coderkun_neo3/keymap.c [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+
+
+// Override Unicode start method to use NE_U instead of KC_U
+void unicode_input_start (void) {
+    register_code(KC_LCTL);
+    register_code(KC_LSFT);
+    register_code(NE_U);
+    unregister_code(NE_U);
+    unregister_code(KC_LSFT);
+    unregister_code(KC_LCTL);
```
[the remark this answers is not in the document]
Comment (coderkun) [reply layouts/community/ergodox/coderkun_neo3/keymap.c]: Fixed.

File: layouts/community/ergodox/coderkun_neo3/keymap.c (hunk 2 above)
Comment (coderkun) [reply layouts/community/ergodox/coderkun_neo3/keymap.c]: Fixed in both keymaps.

**Review by drashna — APPROVED** [review]

File: layouts/community/ergodox/coderkun_neo2/keymap.c [hunk 7]
```diff
@@ -1,116 +1,114 @@
 #include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline layouts/community/ergodox/coderkun_neo2/keymap.c:1]: This file needs a GPLv2-compatible license header.

* [PR Checklist - Example GPLv2 Header - QMK Firmware Documentation](https://docs.qmk.fm/#/pr_checklist?id=example-gplv2-header)

File: layouts/community/ergodox/coderkun_neo3/keymap.c [hunk 8]
```diff
@@ -0,0 +1,288 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline layouts/community/ergodox/coderkun_neo3/keymap.c:1]: This file needs a GPLv2-compatible license header.

* [PR Checklist - Example GPLv2 Header - QMK Firmware Documentation](https://docs.qmk.fm/#/pr_checklist?id=example-gplv2-header)

File: layouts/community/ergodox/coderkun_neo3/keymap.c [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+#include "led.h"
+#include "keymap_extras/keymap_german.h"
+#include "keymap_extras/keymap_neo2.h"
+
+// Layer names
+#define BASE 0      // default layer
+#define FMU  1      // FMU layer
+#define PMN  2      // poor man’s Neo (Neo 2)
+#define PMB  3      // poor man’s Bone (Neo 3)
+#define NHL  4      // Neo/Bone’s software layer 4 rebuilt in Hardware
```
Comment (noroadsleft) [inline layouts/community/ergodox/coderkun_neo3/keymap.c:11]:
```suggestion
enum layer_names {
    BASE,       // default layer
    FMU,        // FMU layer
    PMN,        // poor man’s Neo (Neo 2)
    PMB,        // poor man’s Bone (Neo 3)
    NHL,        // Neo/Bone’s software layer 4 rebuilt in Hardware
};
```

`layouts/community/ergodox/coderkun_neo2/keymap.c` could use a similar treatment as well.

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Comment by coderkun** [comment]
Thanks for your feedback on this PR but I don’t own an Ergodox anymore and therefore won’t work on this layout anymore.

**Pull request closed without merging**